docs/bills-matching.md

```diff
@@ -0,0 +1,139 @@
+* `amountLowerDelta` is `bill.matchingCriterias.amountLowerDelta` if it exists, `options.minAmoutDelta` otherwise
+* `amountUpperDelta` is `bill.matchingCriterias.amountUpperDelta` if it exists, `options.maxAmoutDelta` otherwise
+
+All transactions that matches these two criterias are kept.
+
+### Apply filters to neighboring transactions
+
+There are other filters a transaction must pass to match with a bill.
+
+#### Category
+
+Health bills are a little special because they can only match with health
+transactions. So if the bill is an health bill, we keep only health
+transactions. Otherwise, we keep all but health transactions.
+
+You can pass a `allowUncategorized` option to the algorithm, in which case
+uncategorized transactions will also be able to match bills.
+
+#### Reimbursements
+
+This filter is applied only if we are looking for a debit transactions.
+
+If we try to link a bill as a reimbursement to a debit transaction, we check
+that the bill's amount and transaction's reimbursements sum does not overflows
+the transaction's amount.
+
+#### Label regex
+
+This filter is applied only if we are looking for a credit transaction or if
+the bill is not of health type.
+
+This filter checks if a transaction matches a regex that is built upon the
+bill's properties. Here is how the regexp is built:
+
+* If the bill has a `matchingCriterias.labelRegex` property, this is used
+* Otherwise we try to get the brand corresponding to the bill's vendor in our [brands dictionary](https://github.com/cozy/cozy-banks/blob/master/src/ducks/brandDictionary/brands.json). If it exists, we use its `regexp` property
+* Otherwise, we create a regexp containing the bill's vendor
+* Lastly, if the bill has no vendor, we can't do anything and the filter will be `false` for every transaction and the bill will not match any transaction
+
+### Sort remaining transactions
+
+A score is given to the remaining transactions according to their « distance »
+to the bill's date and amount. The closest one wins and continues through the
+algorithm.
+
+### Link bills to debit and credit transactions
+
+When transactions were found for a given bill, we need to link the bill to
+them, either as a `bill` or as a `reimbursement`.
+
+When a bill matched with a transaction (debit or credit), it is added to the
+transaction's `bills` array.
+
+When a bill matched with a debit and a credit transactions, it is also added to
+the debit transaction's `reimbursements` array.
+
+In both cases, a last safety net is applied by checking if adding the bill to
+the transaction would make the bills sum overflowing the transaction amount or
+not. It it overflows, the bill is not added.
+
+## Bills combinations
+
+Sometimes, bills need to be combined to match with a debit transaction. It is
+mostly the case for health bills. For example, when your paid multiple medical
```

Comment:
s/It is mostly the case for health bills/This is generally the case for health bills.